docs/advanced-table-feature.md

```diff
@@ -0,0 +1,65 @@
+|-------------------|------------------------------------------------------------------|------------------------------------------------|------------------------------------|--------------------------------------------------------|
+| Identity          | `{"strategy":"identity","fieldName":["score"]}`                  | `Transforms.identity("score")`                 | `PARTITION BY score`               | Partition by field `score`                             |
+| Hour              | `{"strategy":"hour","fieldName":["score"]}`                      | `Transforms.hour("score")`                     | `PARTITION BY hour(score)`         | Partition by `hour` function in field `score`          |
+| Day               | `{"strategy":"day","fieldName":["score"]}`                       | `Transforms.day("score")`                      | `PARTITION BY day(score)`          | Partition by `day` function in field `score`           |
+| Month             | `{"strategy":"month","fieldName":["score"]}`                     | `Transforms.month("score")`                    | `PARTITION BY month(score)`        | Partition by `month` function in field `score`         |
+| Year              | `{"strategy":"year","fieldName":["score"]}`                      | `Transforms.year("score")`                     | `PARTITION BY year(score)`         | Partition by `year` function in field `score`          |
+| Bucket            | `{"strategy":"bucket","numBuckets":10,"fieldNames":[["score"]]}` | `Transforms.bucket(10, "score")`               | `PARTITION BY bucket(10, score)`   | Partition by `bucket` function in field `score`        |
+| Truncate          | `{"strategy":"truncate","width":20,"fieldName":["score"]}`       | `Transforms.truncate(20, "score")`             | `PARTITION BY truncate(20, score)` | Partition by `truncate` function in field `score`      |
+| List              | `{"strategy":"list","fieldNames":[["dt"],["city"]]}`             | `Transforms.list(new String[] {"dt", "city"})` | `PARTITION BY list(dt, city)`      | Partition by `list` function in fields `dt` and `city` |
+| Range             | `{"strategy":"range","fieldName":["dt"]}`                        | `Transforms.range(20, "score")`                | `PARTITION BY range(score)`        | Partition by `range` function in field `score`         |
+
+Except the strategies above, you can use other functions strategies to partition the table, for example, the strategy can be `{"strategy":"functionName","fieldName":["score"]}`. The `functionName` can be any function name that you can use in SQL, for example, `{"strategy":"functionName","fieldName":["score"]}` is equivalent to `PARTITION BY functionName(score)` in SQL. 
+For complex function, please refer to `FunctionPartitioningDTO`. 
+
+#### Bucketed table
```

Comment:
What's the bucked table? What's the difference between this and paritioned table?

Reply:
Please see: Hive create table [syntax](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-hiveformat.html) and https://docs.google.com/document/d/1njI1gn-0oedM03NxPG3VpPFfURTMpucvQAh4WvvY-xg/edit#heading=h.gjj3gc0bkh0

Overall, the bucket works in partitioning data (applied to most engines) and is similar to level-two partitioning.

Reply:
Iceberg supports bucket. The bucket for Iceberg is a special transform.

Reply:
I have noticed it. 
<img width="787" alt="image" src="https://github.com/datastrato/gravitino/assets/15794564/049c46af-862f-42f2-be80-a136dc74fcd3">

Not all engines support parititioning, bucket, and sort order at the same time and Iceberg is just among them.